### PR TITLE
fix: migration CLI by making packages external

### LINKED ISSUE #3578: Migration error: Dynamic require of "node:events" is not supported
### Which generators are impacted?

- [ ] All
- [ ] HTML
- [ ] React
- [X] Angular
- [ ] Vue
- [ ] Web components
- [ ] Power Apps

### Reproduction case

When I try to update from version 0.5.6 to version 0.6.0, I receive the following error:

``` 
npx @db-ui/foundations@v0.6 migration --type=v005_v006 --src=./src
file:///Users/dannykoppenhagen/dev/ri-angular/node_modules/@db-ui/foundations/build/index.js:12
  throw Error('Dynamic require of "' + x + '" is not supported');
        ^

Error: Dynamic require of "node:events" is not supported
    at file:///Users/dannykoppenhagen/dev/ri-angular/node_modules/@db-ui/foundations/build/index.js:12:9
    at ../../node_modules/commander/lib/command.js (file:///Users/dannykoppenhagen/dev/ri-angular/node_modules/@db-ui/foundations/build/index.js:965:25)
    at __require2 (file:///Users/dannykoppenhagen/dev/ri-angular/node_modules/@db-ui/foundations/build/index.js:15:50)
    at ../../node_modules/commander/index.js (file:///Users/dannykoppenhagen/dev/ri-angular/node_modules/@db-ui/foundations/build/index.js:3009:33)
    at __require2 (file:///Users/dannykoppenhagen/dev/ri-angular/node_modules/@db-ui/foundations/build/index.js:15:50)
    at file:///Users/dannykoppenhagen/dev/ri-angular/node_modules/@db-ui/foundations/build/index.js:3386:28
    at ModuleJob.run (node:internal/modules/esm/module_job:262:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:475:24)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:109:5)

Node.js v22.2.0
```

### Expected Behaviour

It should work and migrate stuff

### Screenshots

_No response_

### Browser version

None

### Add any other context about the problem here.

_No response_

### PR DESCRIPTION
**Pull request by nmerget**
## Proposed changes

closes #3578 & https://github.com/db-ui/mono/issues/3581

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (fix on existing components or architectural decisions)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

**Comment by mfranzke** [comment]
@nmerget this might only fix https://github.com/db-ui/mono/issues/3581 instead of https://github.com/db-ui/mono/issues/3578, doesn't it? Or do you actually expect both to be fixed?

**Comment by nmerget** [comment]
It should fix both, yes

**Comment by mfranzke** [comment]
> It should fix both, yes

you're the best

**Comment by mfranzke** [comment]
> It should fix both, yes

@nmerget büümm büümmm

```
node:internal/modules/esm/resolve:854
  throw new ERR_MODULE_NOT_FOUND(packageName, fileURLToPath(base), null);
        ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'commander' imported from /Users/maximilianfranzke/.npm/_npx/0d1f7af59eae6113/node_modules/@db-ui/foundations/build/index.js
    at packageResolve (node:internal/modules/esm/resolve:854:9)
    at moduleResolve (node:internal/modules/esm/resolve:927:18)
    at defaultResolve (node:internal/modules/esm/resolve:1169:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:383:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:352:25)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:227:38)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:87:39)
    at link (node:internal/modules/esm/module_job:86:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v20.16.0
```

**Comment by nmerget** [comment]
We need to publish the migration under another name. The problem is that we need `dependencies` from `package.json` to install something like `commander` in `npx`. But we shouldn't install it for `@db-ui/foundations`

File: packages/migration/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+[![code style: prettier](https://img.shields.io/badge/code_style-prettier-ff69b4.svg?style=flat-square)](https://github.com/prettier/prettier)
+[![Conventional Commits](https://img.shields.io/badge/Conventional%20Commits-1.0.0-yellow.svg)](https://conventionalcommits.org)
+[![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square)](https://makeapullrequest.com)
+
+## Usage
+
+We provide a cli tool to auto migrate your source code. Use this command in your repository:
+
+```shell
+npx @db-ui/foundations migration --type XXX --src=./src
```
Comment (mfranzke) [inline packages/migration/README.md:13]:
```suggestion
npx @db-ui/migration --type XXX --src=./src
```

File: packages/migration/package.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "@db-ui/migration",
+  "version": "0.0.0",
+  "type": "module",
+  "description": "Provides basic tokens and assets based on DB UX Design System Core.",
```
Comment (mfranzke) [inline packages/migration/package.json:5]:
```suggestion
  "description": "Migration script for changing your code in between DB UX Design System v3 versions regarding BREAKING CHANGES.",
```

File: packages/migration/src/migration/index.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { globSync } from 'glob';
-import { ReplaceInFileConfig, replaceInFileSync } from 'replace-in-file';
+import { type ReplaceInFileConfig, replaceInFileSync } from 'replace-in-file';
 import type { ReplaceResult } from 'replace-in-file';
```
Comment (mfranzke) [inline packages/migration/src/migration/index.ts:3]:
```suggestion
import type { ReplaceInFileConfig, ReplaceResult } from 'replace-in-file';
import { replaceInFileSync } from 'replace-in-file';
```

wouldn't this look nicer ?

File: packages/migration/README.md [hunk 4]
```diff
@@ -10,7 +10,7 @@
 We provide a cli tool to auto migrate your source code. Use this command in your repository:
 
 ```shell
-npx @db-ui/foundations migration --type XXX --src=./src
+npx @db-ui/migration -t=v005_v006 -s=./src
```
Comment (mfranzke) [inline packages/migration/README.md:13]: I think we should go with clarity over shortness.

File: packages/migration/README.md (hunk 1 above)
Comment (nmerget) [reply packages/migration/README.md]: fixed

File: packages/migration/package.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/db-ui/mono.git"
+  },
+  "license": "DB Design License",
```
[the remark this answers is not in the document]
Comment (nmerget) [reply packages/migration/package.json]: fixed

File: packages/migration/package.json (hunk 2 above)
Comment (nmerget) [reply packages/migration/package.json]: fixed

File: packages/migration/src/migration/index.ts (hunk 3 above)
Comment (nmerget) [reply packages/migration/src/migration/index.ts]: fixed

File: packages/migration/README.md (hunk 4 above)
Comment (nmerget) [reply packages/migration/README.md]: fixed

File: packages/foundations/scripts/local/generate-icon-overview.ts [hunk 6]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
```
Comment (mfranzke) [inline packages/foundations/scripts/local/generate-icon-overview.ts:1]: @nmerget let's not make this too global, see PR https://github.com/db-ui/mono/pull/3589

**Pull request merged**